### PR TITLE
Default estimates for missing stats [#115534065]

### PR DESCRIPTION
**Pull request by xinzweb**
Change behaviour to use default estimates for relation size
for cases when no statistics are available.

This avoid having heavy statstistics computation as part
of query compilation.

Customer can explicitly using ANALYZE to compute the stats
before running queries to use more accurate stats instead
of default statistics.

Currently, the default size of table is:
- Internal table: 100 pages
- External table: 1000 pages

Signed-off-by: Xin Zhang <xzhang@pivotal.io>

### CODE REVIEW

**Comment by xinzweb** [comment]
@oarap @vraghavan78 @atris @jfrank-pivotal @d

**Comment by vraghavan78** [comment]
+1

File: src/backend/optimizer/util/plancat.c [hunk 1]
```diff
@@ -420,43 +420,10 @@ cdb_estimate_rel_size(RelOptInfo   *relOptInfo,
...
-		 * NOTE: External tables should always have >0 values in pg_class
-		 * (created this way). Therefore we should never get here. However, as
-		 * a security measure (if values in pg_class were somehow changed) we
-		 * plug in our 1K pages 1M tuples estimate here as well, and skip
-		 * cdbRelSize as we can't calculate ext table size.
+		 * Put default estimates in case no statistics are available. This saves cost of asking QEs
 		 */
-		if(!RelationIsExternal(rel))
-		{
-		    size = cdbRelSize(rel);
```
Comment (danielgustafsson) [inline src/backend/optimizer/util/plancat.c]: The `size` variable is now unused and needs to be removed as well.

Comment (xinzweb) [inline src/backend/optimizer/util/plancat.c]: Thanks a lot. Fixed.

**Comment by jfrank-pivotal** [comment]
+1

**Pull request merged**